docs/api.yaml

```diff
@@ -0,0 +1,925 @@
+      tags: 
+        - Subscription
+      requestBody:
+        description: >-
+          Subscription name will be used from path parameter. 
+          #TODO: define a different schema for it.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/subscription'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/subscription'
+        default:
+          $ref: '#/components/responses/error_response'
+          
+    delete:      
+      description: >-
+        Delete the specified subscription.
+      tags: 
+        - Subscription
+      responses:
+        '200':
+          description: OK
+        default:
+          $ref: '#/components/responses/error_response'
+  
+  
+  /v1/tenants/{tenant_name}/subscriptions/{sub_name}/state:
+    parameters:
+      - $ref: '#/components/parameters/tenanat_param'
+      - $ref: '#/components/parameters/sub_param'
+      
+    get:
+      description: >-
+       Get the state of the subscription.
+      tags: 
+        - Subscription
+      responses:
+        '200':
+          $ref: '#/components/responses/state_response'
+
+        default:
+          $ref: '#/components/responses/error_response'
+    put:
+      description: >-
+       Perform the specified operation asynchronously to update the Subscription state. 
+       Allowed operations are block, unblock, start and stop.
+      tags: 
+        - Subscription
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/state_request'
+      responses:
+        '200':
+          description: OK
+        default:
+          $ref: '#/components/responses/error_response'
+  
+          
+  /v1/tenants/{tenant_name}/subscriptions/{sub_name}/status:
+    parameters:
+      - $ref: '#/components/parameters/tenanat_param'
+      - $ref: '#/components/parameters/sub_param'
+    get:
+      description: >-
+        Get the current status of this subscription.
+      tags: 
+        - Subscription
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/consumer_status'
+        default:
+          $ref: '#/components/responses/error_response'
+
+
+  /v1/tenants/{tenant_name}/subscriptions/{sub_name}/dlt:
+    parameters:
+      - $ref: '#/components/parameters/tenanat_param'
+      - $ref: '#/components/parameters/sub_param'
+      - $ref: '#/components/parameters/http_code'
+      - $ref: '#/components/parameters/from_param'
+      - $ref: '#/components/parameters/to_param'
+      - $ref: '#/components/parameters/msg_id_param'
+      - $ref: '#/components/parameters/group_id_param'
```

Comment:
group_id vs group_key. we can stick to one.

Reply:
aligned .. any preferences ?